js/app/prospectus-events-filter.js

```diff
@@ -1,458 +1,401 @@
-          '</p>' +
-
-          '<h4 class="events-feed__title">' +
-            event['name'] +
-          '</h4>';
-
-      html += '<p class="events-feed__time">';
-      if (event['time_start']) {
-        html += '<time dateTime="' + formatTime(event["time_start"]) + '">' + formatTime(event["time_start"]) + '</time>';
-      }
-      if (event["time_end"]) {
-        html += ' - ' + '<time dateTime="' + formatTime(event["time_end"]) + '">' + formatTime(event["time_end"]) + '</time>';
-      }
-      if (event['link']) {
-        html += '<div class="forward-link"> <a href="' + event['link'] + '" class="events-feed__title-link">Find out more and book your place</a></div>';
+    return '';
+  }
+
+  function printEventCounter(events) {
+    var showVideos = $("#show-videos").val() == 'on-demand';
+    var live = 0;
+    var onDemand = 0;
+    for (var i = 0; i < events.length; i++) {
+      if (isFutureEvent(events[i])) {
+        live++;
+      } else if (eventHasVideo(events[i])) {
+        onDemand++;
       }
-      html += '</div>' +
-              '</div>';
-      return html;
     }
+    var liveS = (live != 1?"s":"");
+    var onDemandS = (onDemand != 1?"s":"");
+    if (showVideos) {
+      return 'Showing ' + onDemand + ' ' + 'on-demand event' + onDemandS + '; ' + live + ' live event' + liveS + ' also available.';
+    } else {
+      return 'Showing ' + live + ' ' + 'live event' + liveS + '; ' + onDemand + ' on-demand event' + onDemandS + ' also available.';
+    }
+  }
 
-    function printSidebarFutureEvent(event, n) {
-      function formatTime(time) {
-        if (time.length > 5) {
-          return time.substring(0,time.lastIndexOf(":"));
-        }
-        return time;
-      }
-      var timehtml = "";
-      if (event['time_start']) {
-        timehtml = "<br />" + formatTime(event["time_start"]);
-        if (event["time_end"]) {
-          timehtml += " - " + formatTime(event["time_end"]);
-        }
-      }
-      var titlehtml = (event['link']?'<a target="_top" href="' + event['link'] + '" class="clicktracker singleeventclick">' + event['name'] + '</a>':'<span>' + event['name'] + '</span>');
-      var html = '' +
-      '<tr>' +
-        '<td>' +
-          '<h4 class="heading">' +
-            titlehtml +
-          '</h4>' +
-          '<div><strong>Date:</strong> ' +
-            '<span class="fa fa-calendar event-icon">&nbsp;</span>' +
-            '<span class="">' +
-              event['final_date'] + 
-            '</span>' + timehtml +
-            '<br />' +
-            '<strong>Location:</strong> ' + event['location'] + 
-          '</div>' +
-          '<div>' +
-            event['description'] +
-          '</div>' +
-        '</td>' +
-      '</tr>';
-      return html;
+  function printEvent(event, n) {
+    if (isEventsWidget) {
+      return (isFutureEvent(event)?printFutureEvent(event, n):printVideoEvent(event, n));
+    } else {
+      return (isFutureEvent(event));
     }
```

Comment:
This makes little sense. There's no point checking whether it's the Events Widget if there is no sidebar widget. Just do the return on line 74.